aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -101,6 +102,18 @@ class CAFFE2_API OperatorEntry final {
     dispatchTable_.setManuallyBoxedKernel_(func);
   }
 
+  // Asserts that the given FuncType is correct for calling this operator in an unboxed way.
+  template<class FuncType>
+  void assertSignatureIsCorrect() {
+    TORCH_INTERNAL_ASSERT(!cpp_signature_.has_value() || (CppSignature::make<FuncType>() == *cpp_signature_),
+        "Tried to access operator ", name_, " with a wrong signature. Accessed with ",
+        CppSignature::make<FuncType>().name(),
+        " but the operator was registered with ",
+        cpp_signature_->name(),
+        " This likely happened in a call to OperatorHandle::typed<Return (Args...)>(). Please make sure that the function signature matches the signature in the operator registration call."
```

Comment:
Can you print out the `debug_` info here? That will let people know where the operator was registered